litestar/openapi/config.py

```diff
@@ -88,31 +105,111 @@ class OpenAPIConfig:
 
     :class:`Reference <litestar.openapi.spec.reference.Reference>` objects.
     """
-    root_schema_site: Literal["redoc", "swagger", "elements", "rapidoc"] = "redoc"
-    """The static schema generator to use for the "root" path of `/schema/`."""
-    enabled_endpoints: set[str] = field(
-        default_factory=lambda: {
-            "redoc",
-            "swagger",
-            "elements",
-            "rapidoc",
-            "openapi.json",
-            "openapi.yaml",
-            "openapi.yml",
-            "oauth2-redirect.html",
-        }
-    )
-    """A set of the enabled documentation sites and schema download endpoints."""
     operation_id_creator: OperationIDCreator = default_operation_id_creator
     """A callable that generates unique operation ids"""
     path: str | None = field(default=None)
-    """Base path for the OpenAPI documentation endpoints."""
+    """Base path for the OpenAPI documentation endpoints.
+
+    If no path is provided the default is ``/schema``.
+    """
+    render_plugins: Sequence[OpenAPIRenderPlugin] = field(default=())
+    """Plugins for rendering OpenAPI documentation UIs."""
+    openapi_controller: type[OpenAPIController] | None = None
+    """Controller for generating OpenAPI routes.
+
+    Must be subclass of :class:`OpenAPIController <litestar.openapi.controller.OpenAPIController>`.
+    """
+    root_schema_site: Literal["redoc", "swagger", "elements", "rapidoc"] | None = None
+    """The static schema generator to use for the "root" path of `/schema/`."""
+    enabled_endpoints: set[str] | None = None
+    """A set of the enabled documentation sites and schema download endpoints."""
```

Comment:
What about something like
```py
    router_kws: RouterKwsDict = field(default_factory=dict)
    """Keyword arguments passed to the OpenAPI router on instantiation."""
```
This would allow users to define things like guards etc for the router. Thoughts?

Reply:
Why not just allow passing in a router directly? We could still `register` things dynamically on it then at that would make passing the config a bit easier.

Reply:
Added `openapi_router` to the `OpenAPIConfig`.